### PR TITLE
Add mean property predictor

### PR DESCRIPTION
**Pull request by latture**
This PR adds a `MeanPropertyPredictor` and deprecates the `GeneralizedMeanPropertyPredictor`. The former requires that properties are specified as list of `RealDescriptors` instead of a list of descriptor keys. (We use a list of real descriptors and not a set because real descriptors are not hashable, at least not currently.)

### CODE REVIEW

File: src/citrine/informatics/predictors/mean_property_predictor.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+
+class MeanPropertyPredictor(
+        Serializable['MeanPropertyPredictor'], Predictor):
+    """[ALPHA] A predictor interface that computes mean component properties.
+
+    .. seealso::
+       If you are using a deprecated generalized mean property predictor please see
+       :class:`~citrine.informatics.predictors.GeneralizedMeanPropertyPredictor`
+       for details on how to migrate to the new format.
```
Comment (bfolie) [inline src/citrine/informatics/predictors/mean_property_predictor.py:21]: I don't see these details in `GeneralizedMeanPropertyPredictor`

**Comment by bfolie** [comment]
>We use a list of real descriptors and not a set because real descriptors are not washable

Washable?

**Comment by latture** [comment]
> > We use a list of real descriptors and not a set because real descriptors are not washable
> 
> Washable?

Sorry, autocorrect.. that should be hashable.

File: src/citrine/informatics/predictors/mean_property_predictor.py (hunk 1 above)
Comment (latture) [reply src/citrine/informatics/predictors/mean_property_predictor.py:21]: I don't know where those changes went. Hold on while I rewrite it...

Comment (latture) [reply src/citrine/informatics/predictors/mean_property_predictor.py]: Just pushed the docs.

**Review by bfolie — APPROVED** [review]

**Pull request merged**